snapshot/bazel/filer.go

```diff
@@ -5,20 +5,26 @@ import (
 	"github.com/twitter/scoot/snapshot/snapshots"
 )
 
-func MakeDefaultBzFiler() *bzFiler {
+func MakeBzFiler() *bzFiler {
 	return &bzFiler{
-		command: bzCommand{},
+		command: &bzCommand{},
 		updater: snapshots.MakeNoopUpdater(),
 	}
 }
 
-func MakeBzFilerWithLocalStore(localStorePath string) *bzFiler {
-	return &bzFiler{
-		command: bzCommand{
-			localStorePath: localStorePath,
-		},
-		updater: snapshots.MakeNoopUpdater(),
+// Options is a variadic list of functions that take a *bzCommand as an arg
+// and modify its fields, e.g.
+// localStorePath := func(bc *bzCommand) {
+//     bc.localStorePath = "/path/to/local/store"
+// }
+func MakeBzFilerWithOptions(options ...func(*bzCommand)) *bzFiler {
```

Comment:
Have to say I am not a fan of this pattern, strongly prefer explicit parameter list that constructs bzcommand as required. If this was an external API, maybe, but this is a specific interface implementation and not likely to cause breaking changes

Reply:
My hope was to avoid having MakeBzFiler, MakeBzFilerWithLocalStore, MakeBzFilerWithServerAddr, MakeBzFilerWithLocalStoreAndServerAddr, & all the other possible options when skipServer and skipLocalStore are enabled

Reply:
We wouldn't need more than one constructor - MakeBzFiler("", "", false, false) would create the filer with no local store, no server addr, skip options disabled for example.  I will leave this one up to you to change or leave as-is